components/gitlab/new-branch.js

```diff
@@ -0,0 +1,68 @@
+const gitlab = require("https://github.com/PipedreamHQ/pipedream/components/gitlab/gitlab.app.js");
+
+module.exports = {
+  name: "New Branch (Instant)",
+  description: "Emits an event when a new branch is created",
+  version: "0.0.1",
+  props: {
+    gitlab,
+    projectId: { propDefinition: [gitlab, "projectId"] },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
+      const hookParams = {
+        push_events: true,
+        url: this.http.endpoint,
+      };
+      const opts = {
+        hookParams,
+        projectId: this.projectId,
+      };
+      const response = await this.gitlab.createHook(opts);
+
+      const { hookId, token } = response;
+      this.db.set("hookId", hookId);
```

Comment:
I also like to add a little user-facing logging once the webhook is created, just to show them what's going on behind the scenes. You might consider adding something like:

```javascript
console.log(`Webhook created for project ${this.projectId}: ${hookId}`)
```